> I suspect the build failure will be solved by #19902

Only the build compiling with Mono is failing:
> Exts.cs(410,11): error CS1061: 'IEnumerable<TSource>' does not contain a definition for 'TryGetNonEnumeratedCount' and no accessible extension method 'TryGetNonEnumeratedCount' accepting a first argument of type 'IEnumerable<TSource>' could be found (are you missing a using directive or an assembly reference?) [/home/runner/work/OpenRA/OpenRA/OpenRA.Game/OpenRA.Game.csproj]

Does Mono not contain this method?

--- a/OpenRA.Game/Exts.cs
+++ b/OpenRA.Game/Exts.cs
@@ -407,7 +407,8 @@ namespace OpenRA
 
 			// Try to build a dictionary and log all duplicates found (if any):
 			var dupKeys = new Dictionary<TKey, List<string>>();
-			var d = new Dictionary<TKey, TElement>();
+			var capacity = source is ICollection<TSource> collection ? collection.Count : 0;
+			var d = new Dictionary<TKey, TElement>(capacity);
 			foreach (var item in source)
 			{
 				var key = keySelector(item);
@@ -418,7 +419,7 @@ namespace OpenRA
 					continue;
 
 				// Check for a key conflict:
-				if (d.ContainsKey(key))
+				if (!d.TryAdd(key, element))
 				{
 					if (!dupKeys.TryGetValue(key, out var dupKeyMessages))
 					{
@@ -430,10 +431,7 @@ namespace OpenRA
 
 					// Log this conflicting value:
 					dupKeyMessages.Add(logValue(element));
-					continue;
 				}
-
-				d.Add(key, element);
 			}
 
 			// If any duplicates were found, throw a descriptive error

